### DISCUSSION TITLE
repository name must be lowercase

### DISCUSSION THREAD

**Discussion by RolfMoleman** [oxsecurity/megalinter#3151] [Q&A] [answered]
I'm trying to work out if this error is being handled correctly. 

Despite my advice, the Azure DevOps project I'm working on contains spaces and upper-case characters, as does its main repo. So do many more in our tenant. 

For this particular example I've managed to successfully switch to lowercase without affecting the remote. However, removing spaces will also affect the remote which ideally I'd like to avoid due to the number of staff affected. 

Is there a simple way to escape the spaces/non docked friendly characters?

**Comment by nvuillam** [comment]
In which use case do you see such error ? :)

**Reply by RolfMoleman** [reply]
I have a project called "BCA Payments" with the repo called the same.

if I use MegaLinter via docker as per below:

```

   # Run MegaLinter
      - script: |
          docker run \
          --volume $(pwd):/tmp/lint \
          --env AZURE_COMMENT_REPORTER='true' \
          --env GIT_AUTHORIZATION_BEARER=$(System.AccessToken) \
          --env CI=true \
          --env DISABLE_ERRORS=false \
          --env TF_BUILD=true \
          --env SYSTEM_ACCESSTOKEN=$(System.AccessToken) \
          --env SYSTEM_COLLECTIONURI=$(System.CollectionUri) \
          --env SYSTEM_PULLREQUEST_PULLREQUESTID=$(System.PullRequest.PullRequestId) \
          --env SYSTEM_TEAMPROJECT=$(System.TeamProject) \
          --env BUILD_BUILD_ID=$(Build.BuildId) \
          --env BUILD_REPOSITORY_ID=$(Build.Repository.ID) \
          --env MEGALINTER_CONFIG='/repo_template/.config/.mega-linter.yml' \
          --env VALIDATE_ALL_CODEBASE=false \
          oxsecurity/megalinter:latest
        condition: and(succeededOrFailed(),eq(variables['Build.Reason'], 'PullRequest'))
        displayName: "Run incremental MegaLinter scan with pull request comments"
        enabled: ${{ parameters.run_megalinter }}
        name: "megalinter_docker_pr"
        workingDirectory: "$(System.DefaultWorkingDirectory)"

      - script: |
          docker run \
          --volume $(pwd):/tmp/lint \
          --env AZURE_COMMENT_REPORTER='true' \
          --env GIT_AUTHORIZATION_BEARER=$(System.AccessToken) \
          --env CI=true \
          --env DISABLE_ERRORS=true \
          --env TF_BUILD=true \
          --env SYSTEM_ACCESSTOKEN=$(System.AccessToken) \
          --env SYSTEM_COLLECTIONURI=$(System.CollectionUri) \
          --env SYSTEM_TEAMPROJECT=$(System.TeamProject) \
          --env BUILD_BUILD_ID=$(Build.BuildId) \
          --env BUILD_REPOSITORY_ID=$(Build.Repository.ID) \
          --env MEGALINTER_CONFIG='/repo_template/.config/.mega-linter.yml' \
          --env VALIDATE_ALL_CODEBASE=false \
          oxsecurity/megalinter:latest
        condition: and(succeededOrFailed(),ne(variables['Build.Reason'], 'PullRequest'),ne(variables['Build.Reason'], 'Schedule'))
        displayName: "Run incremental MegaLinter scan"
        enabled: ${{ parameters.run_megalinter }}
        name: "megalinter_docker_ci"
        workingDirectory: "$(System.DefaultWorkingDirectory)"

      - script: |
          docker run \
          --volume $(pwd):/tmp/lint \
          --env AZURE_COMMENT_REPORTER='true' \
          --env GIT_AUTHORIZATION_BEARER=$(System.AccessToken) \
          --env CI=true \
          --env DISABLE_ERRORS=false \
          --env TF_BUILD=true \
          --env SYSTEM_ACCESSTOKEN=$(System.AccessToken) \
          --env SYSTEM_COLLECTIONURI=$(System.CollectionUri) \
          --env SYSTEM_TEAMPROJECT=$(System.TeamProject) \
          --env BUILD_BUILD_ID=$(Build.BuildId) \
          --env BUILD_REPOSITORY_ID=$(Build.Repository.ID) \
          --env MEGALINTER_CONFIG='/repo_template/.config/.mega-linter.yml' \
          --env VALIDATE_ALL_CODEBASE=true \
          oxsecurity/megalinter:latest
        condition: and(succeededOrFailed(),eq(variables['Build.Reason'], 'Schedule'))
        displayName: "Run scheduled full MegaLinter scan"
        enabled: ${{ parameters.run_megalinter }}
        name: "megalinter_docker_sch"
        workingDirectory: "$(System.DefaultWorkingDirectory)"

```

i get 

```

docker: invalid reference format: repository name must be lowercase.
See 'docker run --help'.
##[error]Bash exited with code '125'.

```

**Reply by nvuillam** [reply]
I think the issue is in your arguments in docker run , probably some missing quotes or something like that, because the error is returned by docker , and has nothing to do with your own repository name ^^
https://stackoverflow.com/questions/48522615/docker-error-invalid-reference-format-repository-name-must-be-lowercase

**Comment by nvuillam** [comment]
Maybe `--volume '$(pwd):/tmp/lint'` \ ?

**Reply by RolfMoleman** [reply]
@nvuillam Ill try quoting my env vars and see if that alleviates the problem. 

not sure why I didn’t think to try that 🤦 

Thanks.

**Reply by RolfMoleman** [reply, marked as answer]
@nvuillam 

# results of further testing

Spaces/special characters in Azure DevOps project names and/or repo names do cause issues.

However, this is a docker limitation and not an issue with Megalinter itself. Docker complained further about the use of quotes in an Azure DevOps pipeline.

# Solutions

1. The best solution in my opinion - avoid naming projects or repositories with special characters other than . - or _

2. alternatively - make use of pipeline variables to manipulate unsupported characters. However, this will affect the use of the pull request commenter.

**Reply by nvuillam** [reply]
I didn't even know that we could create repository names with spaces, it seems quite unusual ^^

**Reply by RolfMoleman** [reply]
> I didn't even know that we could create repository names with spaces, it seems quite unusual ^^


In GitHub, you can't. It replaces the space with a - automatically.

Azure DevOps, permits spaces.

Not sure if Atlassian or other systems do or not.
